cloud/ec2_client.go

```diff
@@ -207,6 +207,17 @@ func (c *awsClientImpl) TerminateInstances(input *ec2.TerminateInstancesInput) (
 			output, err = c.EC2.TerminateInstances(input)
 			if err != nil {
 				if ec2err, ok := err.(awserr.Error); ok {
+					if strings.Contains(ec2err.Code(), EC2ErrorNotFound) {
+						grip.Debug(message.WrapError(ec2err, message.Fields{
```

Comment:
You should also call host.Terminate here to mark it as terminated in the DB

Reply:
This is the client for talking to AWS. Anything that calls this is responsible for terminating the host in the DB. What this changes is that if a caller asks this client to terminate a host in AWS that no longer exists (e.g., it has been an hour, so AWS has removed the terminated instance from its API), instead of returning an error, we will return nil, because no such host exists, and is therefore terminated.

Reply:
Can you modify the monitor code to set the host as terminated in the DB if this happens then? That will prevent it from continuously trying to terminate hosts that have been reaped. It should be on line 199 of hosts.go

Reply:
Oh, good catch, thank you. Done.